DentOS_Framework/DentOsTestbed/src/dent_os_testbed/utils/test_utils/cleanup_utils.py

```diff
@@ -0,0 +1,108 @@
+from dent_os_testbed.constants import DEFAULT_LOGGER
+from dent_os_testbed.lib.ip.ip_address import IpAddress
+from dent_os_testbed.lib.ip.ip_link import IpLink
+from dent_os_testbed.lib.ip.ip_route import IpRoute
+from dent_os_testbed.lib.tc.tc_qdisc import TcQdisc
+from dent_os_testbed.logger.Logger import AppLogger
+from dent_os_testbed.utils.test_utils.tb_utils import (
+    tb_get_all_devices
+)
+from dent_os_testbed.utils.test_utils.tgen_utils import (
+    tgen_utils_get_dent_devices_with_tgen
+)
+
+
+async def post_test_cleanup_qdiscs(testbed):
+    """Removes all non-default qdiscs created during test. 
+    Can be used separately or by using `cleanup_qdiscs` fixture."""
+    logger = AppLogger(DEFAULT_LOGGER)
+    dev = await tb_get_all_devices(testbed)
+    dev = dev[0]
+    logger.info("Clearing TC")
+    out = await TcQdisc.show(
+        input_data=[{dev.host_name: [{"options": "-j"}]}], 
+        parse_output=True
+    )
+    qdiscs_info = out[0][dev.host_name]["parsed_output"]
+    for qdisc_obj in qdiscs_info:
+        if qdisc_obj["kind"] != "noqueue":
+            await TcQdisc.delete(
```

Comment:
Please form a list of commands firs and then call all of them in a single TcQdisc.delete()

Reply:
The issue with a list of commands is if one command fails then the other commands will not be executed. One such scenario is if we have a root qdisc (ets) and a bunch of leaves (tbf/red). If the root qdisc is deleted first then the leaves will be automatically deleted and the next (and subsequent) `tc qd del` command will fail.